README.md

```diff
@@ -2,20 +2,30 @@
 
 ## Getting Started
 
+### Database
+
 For full functionality, the frontend and backend must both be run in conjunction with a local mongoDB server. Instructions for installing mongoDB can be found [here](https://github.com/SE750-DART/nerds-oppose-society/wiki/Setting-Up-MongoDB).
 
-To start the frontend:
+### Monorepo tooling
 
-```
+This project is managed as a monorepo using [npm workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces). You must initially `npm i` from the root of the project to properly setup tooling, including git hooks through [husky](https://typicode.github.io/husky/#/). Calling `npm i` from the root will install all required packages of `backend/` and `frontend/`. After the initial install from the root you are able to use `npm i` from subdirecties `backend/` and `frontend/` to update and install new packages.
+
+### Frontend
+
+To start:
+
+```sh
 cd frontend/
-npm ci
+npm i
```

Comment:
The `package-lock.json`'s got deleted from `backend/` and `frontend/` as a result of the move to npm workspaces - they're now consolidated in the root `package-lock.json`.

Regarding `package-lock.json` churn I don't think it's that much of an issue - imo it's better to get more regular small changes that keep dependencies up-to-date than remaining stagnant for extended periods of time.

Reply:
Right I guess so - random package-lock.json additions to PRs can be very annoying though

Reply:
Does npm ci work in workspaces?

Reply:
Yep! Also in the GitHub Actions I've used `npm ci -w=backend` - which means it will only install the packages needed in the context of the backend (which speeds things up since we run seperate CI for frontend & backend)

Reply:
I'd still prefer having ci in the readme just for best practice/to reduce churn - mainly in case anyone else jumps in and is inexperienced with the churn package-lock.json introduces and how to manage it. I think Ben said he'd be interested in having a look at some point, especially if we bring this back into active dev

Reply:
On `package-lock.json` I guess another way to put it is that the churn is a bit of a fallacy -if you really wanted to prevent it you'd remove all the [carets (^)](https://docs.npmjs.com/cli/v6/using-npm/semver#caret-ranges-123-025-004) from your dependency versions & then you could just `npm i` as usual. I still think the "churn" is a good thing since it means that backwards-compatible updates for dependencies are making their way to production more regularly. Further evidence is probably found in the [npm docs](https://docs.npmjs.com/cli/v6/commands/npm-ci):

> This command is similar to [npm install](https://docs.npmjs.com/cli/v6/commands/npm-install), except it's meant to be used in automated environments such as test platforms, continuous integration, and deployment...

Reply:
Just chiming in on my phone to say yeah if we want to version lock we should be using absolute versions in the package.json.

Technically absolute versions in combination with a tool like `renovate` is the best way to keep dependencies updated and also ensure all devs run the same version of libraries locally

Reply:
Yeah, I don't think version locking is necessary for us, & by using `npm ci` to prevent `package-lock.json` changes you're only really delaying then to come in a big wave when someone installs/updates a package (`npm i <packageName>`). Something like `renovate` could be good to add regardless so packages are regularly updated & would keep any "churn" from `npm i` minimal.

Reply:
My main issue is that yep when you do `npm i` to install something new, at least you can tag that in a specific commit as the reason for the updated `package-lock.json`. It comes from commits that are like "Change X logic" with a fat `package-lock.json` that is unrelated to the changes. I guess the broader solution would be to write contributing guidelines (that we enforce) saying that a PR will be rejected if there's `package-lock.json` churn, and then maybe using something like dependabot to bump absolute versions in `package.json`. I'm happy to let this PR through and move the discussion to an issue to be resolved down the line though

Reply:
Yeah if you're interested in preventing `package-lock.json` churn I think that could fall under the scope of another PR/GitHub App setup that provides regimented tooling for it - until then I think `npm i` should be the default.

Reply:
I'm just going to be more strict on PR reviews with package-lock churn until then I guess 😂